Directory.Packages.props

```diff
@@ -54,11 +54,9 @@
     -->
     <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="8.0.0" />
 
-    <!-- A conservative version of System.Text.Encodings.Web must be used here since there is no backward compatibility guarantee during major version bumps. -->
```

Comment:
What's the position/response regarding this comment? I don't think we can silently remove it without clarity.

Reply:
It looks like the CVE is only related to `System.Text.Json`. Why are we updating the version of `System.Text.Encodings.Web`? Are we changing our stance on how we select the versions for these packages?

Reply:
> It looks like the CVE is only related to System.Text.Json. Why are we updating the version of System.Text.Encodings.Web? Are we changing our stance on how we select the versions for these packages?

The version 8.0.4 of `System.Text.Json` has a stricter dependency on `System.Text.Encodings.Web` version 8.0.0. This necessitates us to also update `System.Text.Encodings.Web`.

Reply:
> What's the position/response regarding this comment? I don't think we can silently remove it without clarity.

Updated the PR description to explain the reasoning behind it.

Reply:
What happens when `9.0.0` version of `System.Text.Json` is released (without `8.0.4` marked for any vulnerability)? Do we simply bump the version to `9.0.0` or stick to the lowest secure version of the package?

Reply:
A few of the packages are using conservative versions to support auto-instrumentation project. However, now auto-instrumentation or other projects are forced to find different workarounds to overcome this behavior. The answer to this question depends on the needs of other projects. If auto-instrumentation finds a solution to overcome this problem for all future versions, then bumping might not be an issue. Otherwise, we have to be conservative and stick to 8.0.4.

Reply:
It will be great if we can stay on the lowest possible version as long as possible.
In this case, we can upgrade this dependency when .NET8 is not longer supported. During the upgrade, we should consider the lowest version included in any supported .NET version.

As long packages are backward compatible, .NET Framework is no a problem. We are shipping the latest packages for such users..